index.js

```diff
@@ -13,30 +13,66 @@ module.exports = function(callback, options) {
   options = {
     $current: options.$current || 1,
     max: options.max,
-    timeout: options.timeout || undefined
+    timeout: options.timeout || undefined,
+    match: options.match || [],
+    backoffBase: options.backoffBase === undefined ? 1.2 : options.backoffBase,
+    backoffExponent: options.backoffExponent || 1.5
   };
+  
+  // Massage match option into array so we can blindly treat it as such later
+  if (!Array.isArray(options.match)) options.match = [options.match];
 
   debug('Trying '+callback.name+' (%s)', options.$current);
 
 
   return new Promise(function (resolve, reject) {
-    var timeout;
+    var timeout, backoffTimeout;
     if (options.timeout) {
       timeout = setTimeout(function () {
+        if (backoffTimeout) clearTimeout(backoffTimeout);
         reject(Promise.TimeoutError(callback.name + ' timed out'));
       }, options.timeout);
     }
 
     Promise.resolve(callback()).then(resolve).tap(function () {
       if (timeout) clearTimeout(timeout);
+      if (backoffTimeout) clearTimeout(backoffTimeout);
     }).catch(function (err) {
       if (timeout) clearTimeout(timeout);
+      if (backoffTimeout) clearTimeout(backoffTimeout);
 
       error(err && err.toString() || err);
+      
+      // Check if we have retried the max number of times
       if (options.$current === options.max) return reject(err);
+      
+      // Only continue retrying if match is zero length, or we find a match
+      var continueTrying = options.match.length === 0;
+      options.match.forEach(function(match) {
+        if (typeof match === "string" && match === err.toString())
```

Comment:
Might want to check against `err.message` instead, i believe that's how bluebird does it for catch.